### PR TITLE
Change frontend to use redux

### PR DESCRIPTION
**Pull request by heeeyi**

### CODE REVIEW

File: microservices/front-end/src/store/slices/questionsSlice.ts [hunk 1]
```diff
@@ -14,13 +14,28 @@ export const questionsSlice = createSlice({
...
+    ) => {
+      const question = state.find(q => q._id === action.payload.qustionId);
+      if (question) {
+        if (action.payload.action === "add") {
+          question.categories.push(action.payload.category);
+        } else {
+          question.categories = question.categories.filter(cat => cat !== action.payload.category);
+        }
+      }
+    },
```
Comment (rexcyrio) [inline microservices/front-end/src/store/slices/questionsSlice.ts:30]: Instead of passing in `"add"` or `"remove"`, we can have two reducers instead

**Review by rexcyrio — COMMENTED** [review]
Generally, I think you're on the right track in integrating Redux into the front end code

File: microservices/front-end/src/components/fetchData.ts [hunk 2]
```diff
@@ -15,46 +15,45 @@ export const getQuestions = async (): Promise<Question[] | undefined> => {
...
-    const res = await fetch("/api/questions");
-    const existingQuestions: Question[] = await res.json();
-
-    const isDuplicateQuestion = existingQuestions.find(
-      (q) => q.title === newQuestion.title
-    );
+  // Removed the try-catch block
+  // First fetch all questions to check for duplicates
+  const res = await fetch("/api/questions");
+  const existingQuestions: Question[] = await res.json();
```
[the remark this answers is not in the document]
Comment (heeeyi) [reply microservices/front-end/src/components/fetchData.ts:36]: Is it possible that backend question service can add a signal that tells me the question title is duplicated? In this way, I can directly handle the issue when the server returns this code

**Review by rexcyrio — APPROVED** [review]
Slight nitpicks. I'll approve this PR so that Randall can work on the collab page

**Pull request merged**